### PR TITLE
[judgels-dev/judgels#4] add login via email

### PR DESCRIPTION
**Pull request by kevinluvian**
judgels-dev/judgels#4

### CODE REVIEW

File: jophiel-app/src/integTest/java/judgels/jophiel/api/session/SessionServiceIntegrationTests.java [hunk 1]
```diff
@@ -43,4 +44,14 @@
                 .isThrownBy(() -> userService.getMyself(AuthHeader.of(sessionToken)))
                 .withMessageContaining("401");
     }
+
+    @Test void login_with_email() {
```
Comment (fushar) [inline jophiel-app/src/integTest/java/judgels/jophiel/api/session/SessionServiceIntegrationTests.java]: Please just include it inside the previous test. Integration tests are expensive to spin up.

File: jophiel-api/src/main/java/judgels/jophiel/api/session/Credentials.java [hunk 2]
```diff
@@ -6,12 +6,14 @@
 @Value.Immutable
 @JsonDeserialize(as = ImmutableCredentials.class)
 public interface Credentials {
-    String getUsername();
+    String getUsername(); // for legacy session
```
Comment (fushar) [inline jophiel-api/src/main/java/judgels/jophiel/api/session/Credentials.java]: Don't need this, but please also modify the legacy session one

**Comment by fushar** [comment]
Thanks! Pending merge until Raphael is fixed :)

**Comment by kevinluvian** [comment]
🎉 🎉

**Pull request merged**